src/core/geometry/qgsabstractgeometry.cpp

```diff
@@ -245,6 +246,28 @@ void QgsAbstractGeometry::transformVertices( const std::function<QgsPoint( const
   // Ideally this would be pure virtual, but SIP has issues with that
 }
 
+QgsAbstractGeometry::part_iterator QgsAbstractGeometry::parts_end()
```

Comment:
Don't we want these camelcase?

Reply:
I've followed the vertex iterator format here -- which I believe was probably done in this way to follow the std iterator convention. So here I'd rather keep consistency with the vertex iterator which is already part of stable API.

Reply:
Looking at Qt, they use this style whenever they have the text "Provided for STL compatibility." within the API docs. I could imagine to follow their lead. It's a good criteria to decide if it's acceptable in a given place.

I'm a tiny bit afraid that having these methods as precedence in the API in combination with a couple of pythonistas (or std c library devs) working on the code might lead to slowly shifting towards inconsistent style in the codebase.

Reply:
I'm unsure - are you totally against this name, or do you want me to leave it but add the extra note to the dox?

Reply:
Sorry for being unclear.

I'm in of consistency with the vertex iterator and keeping it as is.

And at the same time was off-topicly considering future standards for us to follow to not get trapped in "preference" discussions.
I don't think the extra note would make sense here, `parts_end` is rather _stl style_ than _stl compatibility_.

Reply:
I've added a QGIS 4.0 TODO note to revisit this discussion.